test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -212,34 +218,257 @@ func (s *snapshottableTestSuite) DefineTests(driver TestDriver, pattern testpatt
+
+			// Get node object based on the nodeName
+			node, err := cs.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			// Once we have the node name, exec into the host and check if the physical snapshot exists.
+			hostExec := utils.NewHostExec(f)
+			findPhysicalSnapshot := fmt.Sprintf("ls /var/lib/csi-hostpath-data | grep %s", snapshotHandle)
+			result, err := hostExec.Execute(findPhysicalSnapshot, node)
+			framework.ExpectNoError(err)
+
+			// If we find /var/lib/csi-hostpath-data/<snapshotHandle>.snap, the command will exit with 0.
+			framework.ExpectEqual(result.Code, 0)
+		}
 	})
 }
 
+func getHostpathNodeName(cs clientset.Interface, ns string) (string, error) {
+	pods, err := cs.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, p := range pods.Items {
+		// hostpath pods will all be on the same node
+		if strings.Contains(p.Name, "hostpath") {
+			return p.Spec.NodeName, nil
+		}
+	}
+
+	return "", fmt.Errorf("Failed to find node name for hostpath driver")
+}
+
 // WaitForSnapshotReady waits for a VolumeSnapshot to be ready to use or until timeout occurs, whichever comes first.
-func WaitForSnapshotReady(c dynamic.Interface, ns string, snapshotName string, Poll, timeout time.Duration) error {
+func WaitForSnapshotReady(c dynamic.Interface, ns string, snapshotName string, poll, timeout time.Duration) error {
 	framework.Logf("Waiting up to %v for VolumeSnapshot %s to become ready", timeout, snapshotName)
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
+
+	if successful := WaitUntil(poll, timeout, func() bool {
 		snapshot, err := c.Resource(SnapshotGVR).Namespace(ns).Get(context.TODO(), snapshotName, metav1.GetOptions{})
 		if err != nil {
-			framework.Logf("Failed to get claim %q, retrying in %v. Error: %v", snapshotName, Poll, err)
-			continue
-		} else {
-			status := snapshot.Object["status"]
-			if status == nil {
-				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
-				continue
-			}
-			value := status.(map[string]interface{})
-			if value["readyToUse"] == true {
-				framework.Logf("VolumeSnapshot %s found and is ready after %v", snapshotName, time.Since(start))
-				return nil
-			} else if value["ready"] == true {
-				framework.Logf("VolumeSnapshot %s found and is ready after %v", snapshotName, time.Since(start))
-				return nil
-			} else {
-				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
-			}
+			framework.Logf("Failed to get claim %q, retrying in %v. Error: %v", snapshotName, poll, err)
+			return false
+		}
+
+		status := snapshot.Object["status"]
+		if status == nil {
+			framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
+			return false
+		}
+		value := status.(map[string]interface{})
+		if value["readyToUse"] == true || value["ready"] == true {
```

Comment:
I think we don't need to check value["ready"] any more.  That was in a very old version of the alpha api.

Reply:
Removed ` value["ready"] == true `